oneview/data_source_logical_interconnect.go

```diff
@@ -0,0 +1,435 @@
+// specific language governing permissions and limitations under the License.
+
+package oneview
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceLogicalInterconnect() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceLogicalInterconnectRead,
+
+		Schema: map[string]*schema.Schema{
+			"category": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"consistency_status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"eTag": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enclosure_uris": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"ethernet_settings": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"category": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"created": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"dependent_resource_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"eTag": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"enable_fast_mac_cache_failover": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"enable_igmp_snooping": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"enable_network_loop_protection": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"igmp_idle_timeout_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"interconnect_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"mac_refresh_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"modified": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"fusion_domain_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interconnect_map": {
+				Type:     schema.TypeList,
```

Comment:
can you check if a typeset needs to be used here? And also same with other places having type as typeList

Reply:
MultiLevel schema is not working with TypeSets